packages/snaps-controllers/src/snaps/SnapController.ts

```
@@ -1708,9 +1716,15 @@ export class SnapController extends BaseController<
         `Received invalid snap version range: "${newVersionRange}".`,
       );
     }
-
-    const newSnap = await this.fetchSnap(snapId, newVersionRange);
-    const newVersion = newSnap.manifest.version;
+    const newSnap = await this.#fetchSnap(
+      snapId,
+      location ??
+        this.#detectSnapLocation(snapId, { versionRange: newVersionRange }),
+    );
+    if (newSnap.manifest.result === undefined) {
+      throw new Error(JSON.stringify(newSnap.manifest));
```

Can we throw a meaningful error here?

---

Some debug throws got left in